pkg/build/controller/image_change_controller.go

```diff
@@ -78,12 +76,18 @@ func (c *ImageChangeController) HandleImageRepo() {
 			glog.V(4).Infof("Running build for buildConfig %s in namespace %s", config.Name, config.Namespace)
 			b := buildutil.GenerateBuildFromConfig(config, nil, imageSubstitutions)
 			if err := c.BuildCreator.Create(config.Namespace, b); err != nil {
-				glog.V(2).Infof("Error starting build for buildConfig %v: %v", config.Name, err)
+				return fmt.Errorf("Error starting build for buildConfig %v: %v, will be retried", config.Name, err)
```

Comment:
Higher layer = your retry handler.  That's where you make a decision about retry.  Adding it here too confuses the issue